### ISSUE TITLE
step 3 should count the number of covered columns

### ISSUE THREAD

**Issue by kylemcdonald** [addaleax/munkres-js#5]
see https://github.com/bmc/munkres/issues/16 for full explanation.

in short i suggest changing this chunk here https://github.com/addaleax/munkres-js/blob/master/munkres.js#L395-L397

from:

```js
if (this.marked[i][j] == 1) {
	this.col_covered[j] = true;
	++count;
```

to:

```js
if (this.marked[i][j] == 1 && this.col_covered[j] == false) {
	this.col_covered[j] = true;
	++count;
```

**Comment by addaleax** [comment]
Thanks for bringing this issue up – I have to admit I primarily just translated the python code here, so I don’t have an in-depth understanding of the algorithm itself.

That means it would help me a lot if you could explain what kind of change this is… Is it a performance improvement? A bug fix? Does it change the result for some kind of input? The tests keep passing with that change, so I assume that’s a good sign.

(Also, I’d prefer it to have this change upstreamed first, but that’s not a blocker or anything.)

**Comment by kylemcdonald** [comment]
This should be a bug fix, but I also don't understand the algorithm well enough yet to point out exactly where this would fail. I'm basing this change on adherence to the original tutorial that the upstream code was imitating :)

**Comment by addaleax** [comment]
Applied the upstream changes and released `munkres-js@1.2.2`, thanks for the help!

**Comment by kylemcdonald** [comment]
Cheers! Glad this suggestion is helpful even without a PR. Maintaining OSS
can be a pain, especially when keeping a port up to date. Respect.
